contract/contracts/Voting/Voting.sol

```diff
@@ -0,0 +1,107 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.24;
+
+import '@openzeppelin/contracts-upgradeable/access/OwnableUpgradeable.sol';
+import '@openzeppelin/contracts-upgradeable/utils/PausableUpgradeable.sol';
+import '@openzeppelin/contracts-upgradeable/utils/ReentrancyGuardUpgradeable.sol';
+import "./Types.sol";
+
+contract Voting is OwnableUpgradeable, PausableUpgradeable, ReentrancyGuardUpgradeable {
+
+    mapping(uint256 => Types.Proposal) public proposals;
+    uint256 public proposalCount;
+
+    event ProposalAdded(uint256 indexed proposalId, string title, string description);
+    event DirectiveVoted(address indexed voter, uint256 indexed proposalId, uint256 vote);
+    event ElectionVoted(address indexed voter, uint256 indexed proposalId, address indexed candidateAddress);
+    event ProposalConcluded(uint256 indexed proposalId, Types.Proposal proposal);
+
+    function initialize() public initializer {
+        __Ownable_init(msg.sender);
+    }
+
+   function addProposal(Types.Proposal calldata _proposal) public onlyOwner whenNotPaused {
+        proposals[proposalCount] = _proposal;
+        emit ProposalAdded(proposalCount, _proposal.title,_proposal.description);
+        proposalCount++;
+    }
```

Comment:
```suggestion
   function addProposal(Types.Proposal calldata _proposal) public onlyOwner whenNotPaused {
        proposalCount++;
        proposals[proposalCount] = _proposal;
        emit ProposalAdded(proposalCount, _proposal.title,_proposal.description);
    }
```
it should be like this, so the id starts from 1

Reply:
Alright, done!

Reply:
In Solidity, arrays do not automatically handle gaps in indices when directly assigning values. If you attempt to assign a value to proposals[1] without first assigning a value to proposals[0], you will encounter an error because the array does not yet have an element at index 1.